code/datums/elements/light_blocking.dm

```diff
@@ -25,6 +25,12 @@
 		return
 	for(var/turf/turf_loc as anything in movable_target.locs)
 		turf_loc.remove_opacity_source(target)
+		for(var/atom/elem in turf_loc.contents)
+			for(var/datum/light_source/source in elem.light_sources)
+				if(!source)
+					continue
+				if(!source.effect_str)
+					source.update_corners()
```

Comment:
shouldn't remove_opacity_source handle this already?
similarly, if this is required on removal doesn't that mean Moved() reaction is broken even after this fix?

this smells like something else is broken here

Reply:
`light_source` simply does not update when you close all the tiles around it once, since all the `light_corners` that the source created no longer exist. The simplest solution is to force update the light source if it becomes open again. I can move the check for the presence of a light source in the tile to `recalculate_directional_opacity` and it will probably be more accurate